packages/block-editor/src/store/selectors.js

```diff
@@ -2801,35 +2801,18 @@ export const __unstableGetVisibleBlocks = createSelector(
 );
 
 export function __unstableHasActiveBlockOverlayActive( state, clientId ) {
-	// Prevent overlay on blocks with a non-default editing mode. If the mdoe is
-	// 'disabled' then the overlay is redundant since the block can't be
-	// selected. If the mode is 'contentOnly' then the overlay is redundant
-	// since there will be no controls to interact with once selected.
-	if ( getBlockEditingMode( state, clientId ) !== 'default' ) {
-		return false;
-	}
-
 	// If the block editing is locked, the block overlay is always active.
 	if ( ! canEditBlock( state, clientId ) ) {
 		return true;
 	}
 
-	const editorMode = __unstableGetEditorMode( state );
-
-	// In zoom-out mode, the block overlay is always active for section level blocks.
-	if ( editorMode === 'zoom-out' ) {
-		const sectionRootClientId = getSectionRootClientId( state );
-		if ( sectionRootClientId ) {
-			const sectionClientIds = getBlockOrder(
-				state,
-				sectionRootClientId
-			);
-			if ( sectionClientIds?.includes( clientId ) ) {
-				return true;
-			}
-		} else if ( clientId && ! getBlockRootClientId( state, clientId ) ) {
-			return true;
-		}
+	// Section blocks need to be selected first before being able to select their children.
+	if (
+		isSectionBlock( state, clientId ) &&
+		! isBlockSelected( state, clientId ) &&
+		! hasSelectedInnerBlock( state, clientId, true )
+	) {
```

Comment:
I think this change makes it so that when the workspace (gray area) is clicked and then the canvas is clicked, while editing a page one ends up with the content block selected and a weird toolbar showing:


https://github.com/user-attachments/assets/777a46d8-6565-4cfb-8513-9a73b53e2504

Maybe the removal of the check for `getBlockRootClientId`?